src/main/scala/wdl4s/WdlNamespace.scala

```diff
@@ -111,28 +112,24 @@ case class NamespaceWithWorkflow(importedAs: Option[String],
    * and return the value for storage in the symbol store
    */
   def staticDeclarationsRecursive(userInputs: WorkflowCoercedInputs, wdlFunctions: WdlStandardLibraryFunctions): Try[WorkflowCoercedInputs] = {
-    import scala.collection.mutable
-    val collected = mutable.Map[String, WdlValue]()
-    val allDeclarations = workflow.declarations ++ workflow.calls.flatMap {_.task.declarations}
-
-    val evaluatedDeclarations = allDeclarations.filter {_.expression.isDefined}.map {decl =>
-      val value = decl.expression.get.evaluate(declarationLookupFunction(decl, collected.toMap ++ userInputs), wdlFunctions)
-      collected += (decl.fullyQualifiedName -> value.get)
-      val coercedValue = value match {
-        case Success(s) => decl.wdlType.coerceRawValue(s)
-        case f => f
-      }
-      decl.fullyQualifiedName -> coercedValue
-    }.toMap
 
-    val (successes, failures) = evaluatedDeclarations.partition {case (_, tryValue) => tryValue.isSuccess}
-    if (failures.isEmpty) {
-      Success(successes.map {case (k,v) => k -> v.get})
-    } else {
-      val errors = failures.values.collect { case f: Failure[_] => f.exception.getMessage }
-        // .get because failures is guaranteed to be nonEmpty
-      Failure(new ValidationException("Could not evaluate some declaration expressions", errors.toList.toNel.get))
+    def evalDeclaration(accumulated: Map[String, Try[WdlValue]], current: ScopedDeclaration): Map[String, Try[WdlValue]] = {
+      current.expression match {
+        case Some(expr) =>
+          val successfulAccumulated = accumulated.collect({case (k, v) if v.isSuccess => k -> v.get })
+          val value = expr.evaluate(declarationLookupFunction(current, successfulAccumulated ++ userInputs), wdlFunctions)
+          accumulated + (current.fullyQualifiedName -> value)
+        case None => accumulated
+      }
     }
+
+    val workflowStaticDeclarations = workflow.scopedDeclarations.foldLeft(Map.empty[String, Try[WdlValue]])(evalDeclaration)
+
+    val callStaticDeclarations = workflow.calls flatMap { call =>
+      call.scopedDeclarations.foldLeft(Map.empty[String, Try[WdlValue]])(evalDeclaration)
+    } toMap
+
+    TryUtil.sequenceMap(workflowStaticDeclarations ++ callStaticDeclarations)
```

Comment:
> I've also banged my head sometimes coming up with the proper way to use the cumbersome library

I hear ya.  I'm currently in the head-banging mode right now.  I'll see if I can get access to that book.  I'm surprised there aren't better online resources.

So basically right now I'm stuck.  I can sink a hours or a day into figuring out how to make this work with scalaz, but unless somebody wants to sit side-by-side, I'm going to give up in about 15 more minutes.

Here's some work-in-progress code: https://gist.github.com/scottfrazer/25781962cc85276b5bd9